lib/stacks/iam.ts

```diff
@@ -195,23 +201,23 @@ export class IAMStack extends Stack {
                 }),
                 // Endpoint Configuration Permissions
                 new PolicyStatement({
-                    effect: Effect.ALLOW,
+                    effect: Effect.DENY,
```

Comment:
Need to make updates to the [docs](https://github.com/awslabs/mlspace/blob/main/frontend/docs/admin-guide/install.md?plain=1#L213-L225) for these changes.

This is actually going to complicate user installation, because these shouldn't be changed if the customer is not using dynamic roles (most are though)

If not using dynamic roles (few users): Use `Effect.ALLOW` with previous permissions
If using dynamic roles (`MANAGE_IAM_ROLES`): Use these `Effect.DENY` statements

In the docs we will have to provide clear instructions to installers on how to apply the appropriate permissions.

Reply:
Based on what Doug mentioned earlier I added a task for documenting this. This will be hard to explain simply but great callout!

Reply:
In theory, the original `Effect.ALLOW` style permissions should be applied if `MANAGE_IAM_ROLES` is false, but this is only for the case that someone has the permissions to do a CDK deploy that will create these IAM roles and policies and then is intentionally turning off `MANAGE_IAM_ROLES`. I think this should never be a scenario that happens, but is a possible edge-case.

I won't block on this though, because this is a huge anti-pattern no one should probably do